games-board/mah-jong/mah-jong-1.16.ebuild

```diff
@@ -0,0 +1,51 @@
+# Copyright 1999-2020 Gentoo Authors
+# Distributed under the terms of the GNU General Public License v2
+
+EAPI=7
+
+inherit desktop toolchain-funcs xdg
+
+MY_P="mj-${PV}-src"
+DESCRIPTION="A networked Mah Jong program, together with a computer player"
+HOMEPAGE="https://mahjong.julianbradfield.org/"
+SRC_URI="http://mahjong.julianbradfield.org/Source/${MY_P}.tar.gz"
+
+LICENSE="GPL-2"
+SLOT="0"
+KEYWORDS="~amd64 ~x86"
+IUSE=""
+
+RDEPEND="x11-libs/gtk+:2"
+DEPEND="
+	${RDEPEND}
+	dev-lang/perl
+"
+BDEPEND="virtual/pkgconfig"
+
+S=${WORKDIR}/${MY_P}
+
+src_prepare() {
+	default
+
+	sed -i -e '/^.TH/ s/1/6/' xmj.man || die
+	sed -i \
+		-e "/^DESTDIR =/ s:=.*:= ${D}:" \
+		-e "/^BINDIR =/ s:=.*:= /usr/bin:" \
```

Comment:
Shouldn't this have `${EPREFIX}`?

Reply:
${EPREFIX} is empty here. Should I add it anyway?

Reply:
It's not empty only on Gentoo Prefix: https://wiki.gentoo.org/wiki/Project:Prefix/Technical_Documentation#EPREFIX_variable

Reply:
If I replace`-e "/^BINDIR =/ s:=.*:= /usr/bin:" \` with `-e "/^BINDIR =/ s:=.*:= ${EPREFIX}/usr/bin:" \`
it installs the same path `/usr/bin`

If replace with `-e "/^BINDIR =/ s:=.*:= ${EPREFIX}:" \` I get warning 
```
 * The ebuild is installing to one or more unexpected paths:
 * 
 *   /mj-player
 *   /mj-server
 *   /xmj

```

Reply:
Is EPREFIX for configure phase?
Should I then use prefix.eclass for prepare phase?

Reply:
No. If the full path is e.g. /home/user/gentoo/usr/bin, then EPREFIX part of it is `/home/user/gentoo`. Usually it's empty

Reply:
add ${EPREFIX} before /usr/bin

Reply:
I haven't seen the file involved here but the `hprefixify` function from `prefix.eclass` is nicer than manually `sed`ing `EPREFIX` into paths like this.

Reply:
The initial line to be replaced is `BINDIR = bin`.
Do you mean besides sed with `-e "/^BINDIR =/ s:=.*:= /usr/bin:"` 
(and instead of `-e "/^BINDIR =/ s:=.*:= ${EPREFIX}/usr/bin:"`)
inherit `prefix` and then after sed expression I need to add `hprefixify -w /^BINDIR/ Makefile`?

Reply:
Ah, I see it won't work in this case because it relies on the value being prefixed with `/`. It would have been okay if there was a `PREFIX` variable in the Makefile but they've just abused `DESTDIR` for this purpose. That sucks. You could probably avoid all this hassle by just doing this. I haven't tested it though!

```bash
src_compile() {
	emake \
		CC="$(tc-getCC)" \
		EXTRA_CFLAGS="${CFLAGS}" \
		EXTRA_LDOPTIONS="${LDFLAGS}" \
		CDEBUGFLAGS= \
		TILESETPATH="${EPREFIX}/usr/share/${PN}"
}

src_install() {
	emake \
		DESTDIR="${ED}/usr/"
		MANDIR=man/man6 \
		MANSUFFIX=6 \
		INSTPGMFLAGS= \
		install install.man

	…
}
```

Reply:
That would work if they used `?=` to set these variables inside Makefile, but they used `=`. Perhaps need to add a patch (or sed?) which removes `CC=gcc` etc from Makefile, then such emake call should work.

Reply:
Nah, that's not how the rule works. `?=` will prefer an environment variable if it is set but `=` can still be overridden by passing the variables as arguments.

Reply:
@chewi thank you for tip.

`TILESETPATH='"${EPREFIX}/usr/share/${PN}"' ` should be (add outer single parentheses) otherwise it passed as value/expression instead of `char*` and fail to compile.

Now I need fix `install.man` part. Currently it results in sandbox fails trying install into /usr/local. Hope fix it soon.

Reply:
But if you pass it like that, `${EPREFIX}` will be compiled into the binary as is, instead of expanding as a bash variable

Reply:
:(
I will recheck this as somewhere I saw unexpanded values in the begin of build log  while on compilile flags they was expanded in `-DTILESETPATH`.

Reply:
Or try: `TILESETPATH="\"${EPREFIX}/usr/share/${PN}\""`

Reply:
Done